modules/apps/transfer/internal/convert/convert.go

```diff
@@ -0,0 +1,57 @@
+package convert
+
+import (
+	"strings"
+
+	"github.com/cosmos/ibc-go/v8/modules/apps/transfer/internal/denom"
+	v1types "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+	v3types "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types/v3"
+)
+
+// PacketDataV1ToV3 converts a v1 (ICS20-V1) packet data to a v3 (ICS20-V2) packet data.
+func PacketDataV1ToV3(packetData v1types.FungibleTokenPacketData) v3types.FungibleTokenPacketData {
+	if err := packetData.ValidateBasic(); err != nil {
+		panic(err)
+	}
+
+	v2Denom, trace := extractDenomAndTraceFromV1Denom(packetData.Denom)
+	return v3types.FungibleTokenPacketData{
+		Tokens: []*v3types.Token{
+			{
+				Denom:  v2Denom,
+				Amount: packetData.Amount,
+				Trace:  trace,
+			},
+		},
+		Sender:   packetData.Sender,
+		Receiver: packetData.Receiver,
+		Memo:     packetData.Memo,
+	}
+}
+
+// extractDenomAndTraceFromV1Denom extracts the base denom and remaining trace from a v1 IBC denom.
+func extractDenomAndTraceFromV1Denom(v1Denom string) (string, []string) {
+	v1DenomTrace := v1types.ParseDenomTrace(v1Denom)
+
+	splitPath := strings.Split(v1Denom, "/")
+	pathSlice, _ := denom.ExtractPathAndBaseFromFullDenom(splitPath)
```

Comment:
This logic is already done in `ParseDenomTrace`. Instead of repeating the logic, we should probably just trim suffix `splitPath` using `v1DenomTrace.BaseDenom`